app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -52,7 +62,12 @@ def reset_attempt_count_if_user_no_longer_locked_out
 
     UpdateUser.new(
       user: current_user,
-      attributes: { second_factor_attempts_count: 0, second_factor_locked_at: nil }
+      attributes: {
+        second_factor_attempts_count: 0,
+        second_factor_locked_at: nil,
+        otp_send_count: 0,
+        otp_first_sent_at: nil,
```

Comment:
This `reset_attempt_count_if_user_no_longer_locked_out` method only gets called after a user submits an OTP. It's a separate event than the one we are dealing with here. We need a new method in the `OtpLimiter` class that resets `otp_send_count` and `otp_first_sent_at` when the `findtime` has expired. Every time a user is about to be sent an OTP, the questions we need to ask are:

- Has it been more than `findtime` minutes since the user last received an OTP? 
  - YES: set `otp_send_count` to 1 and `otp_first_sent_at` to Time.now. Incidentally, this should probably be called `otp_last_sent_at`.
  - NO: Is `otp_send_count` >= `maxretry`?
    - YES: Prevent the user from receiving OTPs for `bantime` minutes
    - NO: Increment `otp_send_count`